common/errors.go

```diff
@@ -938,7 +935,10 @@ func (e *ErrUpstreamsExhausted) DeepestMessage() string {
 }
 
 func (e *ErrUpstreamsExhausted) UpstreamId() string {
-	return e.DeepSearch("upstreamId")
+	if val := e.DeepSearch("upstreamId"); val != nil {
+		return val.(string)
```

Comment:
To avoid unexpected panics maybe:
```suggestion
		if s, ok := val.(string); ok { return s }
```
?